felix/bpf/ut/precompilation_test.go

```diff
@@ -27,80 +27,92 @@ import (
 	"github.com/vishvananda/netlink"
 
 	"github.com/projectcalico/calico/felix/bpf"
+	"github.com/projectcalico/calico/felix/bpf/bpfutils"
 	"github.com/projectcalico/calico/felix/bpf/tc"
 )
 
+func checkBTFEnabled() []bool {
+	if bpfutils.BTFEnabled {
+		return []bool{false, true}
+	}
+	return []bool{false}
+}
+
 func TestPrecompiledBinariesAreLoadable(t *testing.T) {
 	RegisterTestingT(t)
 
 	bpffs, err := bpf.MaybeMountBPFfs()
 	Expect(err).NotTo(HaveOccurred())
 	Expect(bpffs).To(Equal("/sys/fs/bpf"))
 
+	fmt.Println(checkBTFEnabled())
 	for _, logLevel := range []string{"OFF", "INFO", "DEBUG"} {
 		logLevel := logLevel
 		// Compile the TC endpoint programs.
 		logCxt := log.WithField("logLevel", logLevel)
-		for _, epToHostDrop := range []bool{false, true} {
-			epToHostDrop := epToHostDrop
-			logCxt = logCxt.WithField("epToHostDrop", epToHostDrop)
-			for _, fibEnabled := range []bool{false, true} {
-				fibEnabled := fibEnabled
-				logCxt = logCxt.WithField("fibEnabled", fibEnabled)
-				epTypes := []tc.EndpointType{
-					tc.EpTypeWorkload,
-					tc.EpTypeHost,
-					tc.EpTypeTunnel,
-					tc.EpTypeWireguard,
-				}
-				for _, epType := range epTypes {
-					epType := epType
-					logCxt = logCxt.WithField("epType", epType)
-					if epToHostDrop && epType != tc.EpTypeWorkload {
-						log.Debug("Skipping combination since epToHostDrop only affect workloads")
-						continue
+		for _, btfEnabled := range checkBTFEnabled() {
+			bpfutils.BTFEnabled = btfEnabled
```

Comment:
Cool, LGTM, can we just do `defer bpfutils.BTFEnabled = bpfutils.SupportsBTF()` at the top of the test not to possibly pollute tests that follow?